truth_set/python/structure/solution-approach/standard-method/OrderProcessor.py

```diff
@@ -0,0 +1,19 @@
+class OrderProcessor:
+    def apply_discounts(self, order_total, discounts_applied):
+        if discounts_applied > 5:
+            raise Exception("Too many discounts applied.")
+            
+        if order_total > 100:
+            order_total -= 10  
+        elif order_total > 50:
+            order_total -= 5
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Use constants and custom exceptions</b></div>
<div id="fix">

Consider replacing magic numbers with named constants and creating custom exception class.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1,9 +1,16 @@
 +class TooManyDiscountsError(Exception):
 +    """Raised when too many discounts are applied to an order."""
 +    pass
 +
 +MAX_DISCOUNTS = 5
 +LARGE_ORDER_THRESHOLD = 100
 +MEDIUM_ORDER_THRESHOLD = 50
 +LARGE_ORDER_DISCOUNT = 10
 +MEDIUM_ORDER_DISCOUNT = 5
 +
  class OrderProcessor:
 -    def apply_discounts(self, order_total, discounts_applied):
 -        if discounts_applied > 5:
 -            raise Exception("Too many discounts applied.")
 -        if order_total > 100:
 -            order_total -= 10  
 -        elif order_total > 50:
 -            order_total -= 5
 +    def apply_discounts(self, order_total: float, discounts_applied: int) -> float:
 +        if discounts_applied > MAX_DISCOUNTS:
 +            raise TooManyDiscountsError("Maximum number of discounts exceeded")
 +        if order_total > LARGE_ORDER_THRESHOLD:
 +            order_total -= LARGE_ORDER_DISCOUNT
 +        elif order_total > MEDIUM_ORDER_THRESHOLD:
 +            order_total -= MEDIUM_ORDER_DISCOUNT
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/6#issuecomment-2557204513>#a9a025</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged